src/clr-angular/data/datagrid/render/header-renderer.ts

```diff
@@ -62,12 +71,13 @@ export class DatagridHeaderRenderer implements OnDestroy {
                 this.columnResizer.columnResizeBy = 0;
                 this.widthSet = false;
             }
+            // Don't set width if there is a user-defined one. Just add the strict width class.
             this.renderer.addClass(this.el.nativeElement, STRICT_WIDTH_CLASS);
-            // We don't actually set the width if there already is a user-defined one, we just add the class
             return;
```

Comment:
This is here because we don't want the code outside the if block to run when there is a strict width. We are handling three use cases here:
1. Strict width and user resized to set the width
2. user defined strict width on the column via attr or css
3. columns and cells need to flex and adjust width.